Add new API and speedup notice sending.

vmihailenco opened the pull request:
```
benchmark                 old ns/op     new ns/op     delta
BenchmarkSendNotice-4     32225         31622         -1.87%

benchmark                 old allocs     new allocs     delta
BenchmarkSendNotice-4     122            120            -1.64%

benchmark                 old bytes     new bytes     delta
BenchmarkSendNotice-4     8586          7739          -9.86%
```

zefer commented on notifier.go:
```diff
@@ -152,15 +174,67 @@ func (n *Notifier) SendNotice(notice *Notice) (string, error) {
+
+// NotifyPanic notifies Airbrake about panic and should be used
+// with defer statement.
+func (n *Notifier) NotifyPanic() {
```
Maybe `NotifyIfPanic` would be more descriptive?

zefer commented on README.md:
```diff
@@ -23,13 +23,8 @@ func init() {
-		}
-	}()
-	defer airbrake.Flush()
+	defer airbrake.WaitAndClose(5*time.Second)
```
I like this!

zefer commented:
Do you have benchmarks to share?

zefer commented:
> Add new API

Can you please point this out?

zefer commented on notifier.go:
```diff
@@ -75,6 +87,9 @@ func NewNotifier(projectId int64, projectKey string) *Notifier {
 		n.context["rootDirectory"] = wd
 	}
+	for i := 0; i < 10; i++ {
+		go n.worker()
```
Am I correct in seeing that this configures the notifier with a channel to buffer up to 1000 notices, with 10 workers processing (POSTing) them? What happens when the channel is full and we get more notices?

vmihailenco replied:
It is ignored, see https://github.com/airbrake/gobrake/blob/feature/notify-and-panic/notifier.go#L179-L184

zefer replied:
Thanks.

vmihailenco commented:
Added benchmark to the PR description, but it is too noisy. Also renamed NotifyPanic to NotifyOnPanic.

>Can you please point this out?

Point what? I added here `WaitAndClose` and `NotifyOnPanic`.
